### PR TITLE
OCPBUGS-13034: Cluster-api SA can't create events

### PR DESCRIPTION
**Pull request by jparrill**
**Which issue(s) this PR fixes**
Fixes #[OCPBUGS-13034](https://issues.redhat.com/browse/OCPBUGS-13034)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Review by bryan-cox — APPROVED** [review]
/lgtm

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -2733,6 +2733,11 @@ func reconcileCAPIManagerClusterRole(role *rbacv1.ClusterRole) error {
 			Resources: []string{"customresourcedefinitions"},
 			Verbs:     []string{"get", "list", "watch"},
 		},
+		{
+			APIGroups: []string{""},
```
[the remark this answers is not in the document]
Comment (enxebre) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:2737]: This doesn't make sense to me.
The message you are sharing depends on the permissions the HO has. Being more conservative and removing this perms from the capi cluster role should not be the thing triggering that message.

**Pull request merged**